akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -1639,16 +1647,20 @@ private[akka] class DDataShardCoordinator(
     getShardHomeRequests += (sender -> request)
   }
 
-  private def unstashGetShardHomeRequests(): Unit = {
-    getShardHomeRequests.foreach {
-      case (originalSender, request) => self.tell(request, sender = originalSender)
+  override protected def unstashGetShardHomeRequests(): Unit = {
+    if (getShardHomeRequests.nonEmpty) {
+      // unstash one, will continue unstash of next after receive GetShardHome or update completed
+      val (originalSender, request) = getShardHomeRequests.head
+      self.tell(request, sender = originalSender)
+      getShardHomeRequests -= (originalSender -> request)
```

Comment:
I'm uncomfortable with relying on that `head` and `drop(1)` is the same element in an unordered Set.
but, sure I can capture the tuple and use that instead of creating a new instance.

Reply:
I didn't think on the fact that it's a Set. I still find that find it should be safe. 

That brings us to another question, before it didn't matter to have a Set, but since know we are unstashing one by one, should we respect the ordering? FIFO?

Reply:
Order doesn't matter, aside from possibly fairness between different requests. It wasn't ordered FIFO previously either.